api/group/controller/update.js

```diff
@@ -0,0 +1,19 @@
+import Joi from 'joi';
+import updateGroup from '../service/update.js';
+
+const groupSchema=Joi.object({
+  id: Joi.number().integer().required(),
+  name: Joi.string().required(),
+  permissions:Joi.string().valid('READ','WRITE','DELETE','UPLOAD_FILES','SHARE').required()
```

Comment:
Move permission list to api/group/common/constants.js.

change validation of permissions field to
permissions: Joi.string().valid(...permissions).required()